### DISCUSSION TITLE
Metrics and/or Data Documentation Visible to Developers and End Users

### DISCUSSION THREAD

**Discussion by mdp1125** [evidence-dev/evidence#2656] [Ideas]
### Description

From [this slack thread](https://evidencedev.slack.com/archives/C023MMFSWKU/p1729030541379389)

The TL;DR is that developers and end users can see how data and SQL powering the Evidence app is defined. Like Looker's ? info icon that showed SQL and user defined descriptions on hover in the interface.

### What problem would this solve?

It would save a lot of time, effort, and accuracy if developers and end users could see how data is defined from the interface instead of either not knowing or having to ask around. For example, if in the data warehouse sql some label bucketing column values is defined, or the SQL powering an Evidence chart has a where clause, the end user is not going to know what goes into each bucket or chart filter unless the developer explains so in the Evidence markdown (or the end user is savvy enough to view all the sql queries which may not be realistic). However, any hard coding of column explanations would mean the Evidence app is effectively outdated as soon as it is deployed, so ideally there is something more 'programmatic' in this sense.

### How should it work?

 It would be great to be able to hover over a column in Evidence and see some kind of lineage or data explanation be presented. For example if a column is defined with a case statement and the sql powering it is something like `when column_a in ('x', 'y', 'z') then 'label_i' else 'label_j' end` then the end user should be able to view that quickly without having to go digging through the view SQL queries themselves.

I find myself wanting something like dare I say a LookML-like semantic layer...or at the very least the ability to write a piece of logic and reuse it like a dbt macro i.e. define a 'data_labelled' constant in a way that can write the sql and also the markdown using the same piece of reusable code like a variable. Then the sql and the markdown will always be linked.
